pinot-common/src/main/java/org/apache/pinot/common/utils/CommonConstants.java

```diff
@@ -192,6 +192,9 @@ public ServerType getServerType() {
     public static final String CONFIG_OF_REQUEST_HANDLER_FACTORY_CLASS = "pinot.server.requestHandlerFactory.class";
     public static final String CONFIG_OF_NETTY_PORT = "pinot.server.netty.port";
     public static final String CONFIG_OF_ADMIN_API_PORT = "pinot.server.adminapi.port";
+    // A logical instance id is one which does not contain server host name and/or port info. E.g., server1.
+    // It is by default disabled.
+    public static final String CONFIG_OF_USE_LOGICAL_INSTANCE_ID = "pinot.server.logical.instance.id.enabled";
```

Comment:
Why do we need this ? I think we will always use the Zk info to look up host and port (whether logical id is enabled or not).

Reply:
This config flag is used only for server start and  mainly used for backward compatibility. It controls if we want to overwrite the host/port info in zk using those found server config. The current behavior is to extract hostname from instance id if instance id is found in server config. We do not want to break these use cases for now.